perses/tests/test_repex.py

```diff
@@ -182,11 +185,243 @@ def test_RESTCapableHybridTopologyFactory_repex_charge_mutation():
+                htf = RESTCapableHybridTopologyFactory(
+                    topology_proposal=topology_proposal,
+                    current_positions=old_positions,
+                    new_positions=new_positions
+                )
+
+                # Make sure LRC is set correctly
+                hybrid_system = htf.hybrid_system
+                force_dict = {force.getName(): index for index, force in enumerate(hybrid_system.getForces())}
+                custom_force = hybrid_system.getForce(force_dict['CustomNonbondedForce_sterics'])
+                nonbonded_force = hybrid_system.getForce(force_dict['NonbondedForce_sterics'])
+                custom_force.setUseLongRangeCorrection(False)
+                nonbonded_force.setUseDispersionCorrection(True)
+
+                # Create a reporter
+                selection = "not water"
+                reporter_file = os.path.join(temp_dir, f"{ligand_A_index}_{ligand_B_index}_{phase}.nc")
+                reporter = MultiStateReporter(
+                    reporter_file,
+                    analysis_particle_indices=htf.hybrid_topology.select(selection),
+                    checkpoint_interval=100)
+
+                # Build the hybrid repex sampler
+                sampler = HybridRepexSampler(
+                    mcmc_moves=mcmc.LangevinDynamicsMove(
+                        timestep=4.0 * unit.femtoseconds,
+                        collision_rate=1.0 / unit.picosecond,
+                        n_steps=250,
+                        reassign_velocities=False,
+                        n_restart_attempts=20,
+                        constraint_tolerance=1e-06),
+                    replica_mixing_scheme='swap-all',
+                    hybrid_factory=htf,
+                    online_analysis_interval=None)
+
+                sampler.setup(
+                    n_states=n_states,
+                    temperature=300 * unit.kelvin,
+                    storage_file=reporter,
+                    minimisation_steps=0,
+                    endstates=True)
+
+                sampler.energy_context_cache = cache.ContextCache(capacity=None, time_to_live=None, platform=platform)
+                sampler.sampler_context_cache = cache.ContextCache(capacity=None, time_to_live=None, platform=platform)
+
+                # Run repex
+                sampler.extend(n_iterations)
+
+                # Retrieve free energy
+                reporter.close()
+                reporter = MultiStateReporter(reporter_file)
+                analyzer = MultiStateSamplerAnalyzer(reporter, max_n_iterations=n_iterations)
+                # f_ij, df_ij = analyzer.get_free_energy()
+                # Extract uncorrelated energy matrix (u_ln) and samples from states (N_l)
+                energy_matrix = analyzer._unbiased_decorrelated_u_ln
+                sampled_states = analyzer._unbiased_decorrelated_N_l
```

Comment:
we actually want to feed MBAR the unsampled states as well, which is why I normally compute the reduced potential matrix (aka uncorrelated energy matrix) like this: https://github.com/choderalab/perses-barnase-barstar-paper/blob/main/scripts/05_analyze/analysis_tools.py#L104
This is minor and shouldn’t affect the test results, since that unsampled endstate effect should cancel out when computing the DDG. But I think we want to include the unsampled endstates here to make sure that its testing the "normal" pipeline for repex.

Reply:
I think you mean decorrelated samples instead of unsampled states? Or what's the difference here? Please check the changes in https://github.com/choderalab/perses/pull/1065/commits/59dbceae8ccc8161a502f07e4de6ef52b122b897 and let me know if this works, thanks!